File: 数据存储/CoreDatatest/CoreDatatest/ViewController.m
//
//  ViewController.m
//  CoreDatatest
//
//  Created by YZH on 2018/2/14.
//  Copyright © 2018年 YZH. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib
    AppDelegate *app = [[AppDelegate alloc] init];
   [app insertStudent];
    [app selectStudent];
   // [app deleteStudent];
    [app updateStudent];
}

//-(void)insertStudent{
//    NoteManagerObject
//}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: UI/testfield/testfield/ViewController.swift
//
//  ViewController.swift
//  testfield
//
//  Created by YZH on 2018/1/15.
//  Copyright © 2018年 YZH. All rights reserved.
//

import UIKit

class ViewController: UIViewController,UITextFieldDelegate{

    @IBOutlet weak var textfield: UITextField!
    override func viewDidLoad() {
        super.viewDidLoad()
        //self.textfield.delegate=self
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    func textFieldShouldBeginEditing(_ textField: UITextField) -> Bool {
        print("call textFieldShouldBeginEditing")
        return true
    }
    
    func textFieldDidBeginEditing(_ textField: UITextField) {
        print("call textFieldDidBeginEditing")
    }

    func textFieldShouldEndEditing(_ textField: UITextField) -> Bool {
        print("call textFieldShouldEndEditing")
        return true
    }
    
    func textFieldDidEndEditing(_ textField: UITextField) {
        print("call textFieldDidEndEditing")
    }
    
    func textFieldShouldReturn(_ textField: UITextField) -> Bool {
        print("call textFieldShouldReturn")
        textfield.resignFirstResponder()
        return true
    }
    
}

